Add gRPC 1.4 and Python 3.6 support

davidgasquez opened the pull request:
This PR is related to https://github.com/grpc/grpc-docker-library/issues/45. The new `1.4` folder contains a Dockerfile that implements gRPC 1.4 for Python 3.6.

Happy to change anything! Let me know your thoughts and feedback 😄

davidgasquez commented:
I signed it!

nathanielmanistaatgoogle commented:
Please squash commits and [conform to these commit message guidelines](https://chris.beams.io/posts/git-commit/#seven-rules).

I'd forgotten this repository existed! What brings you here? Thanks for the improvement!

davidgasquez commented:
Commits squashed @nathanielmanistaatgoogle! Let me know if I should do anything else :) 

We've started to use gRPC and instead of building our own base image I decided to contribute here. 😄

nathanielmanistaatgoogle commented:
Whoops; something else I forgot: [we recently relicensed](https://github.com/grpc/grpc/commit/7897ae9308a25a064ee2e7386da25d37cc3273fa), so I don't think it's appropriate to copy the license headers off the extant files in this repository. See [this file](https://github.com/grpc/grpc/blob/e313a68e1ffb42b784f75bab9a7023cb145d58cc/tools/dockerfile/grpc_clang/Dockerfile#L1-L13) for the license header that (I believe) should be used for new files.

davidgasquez committed:
Add Python 3.6 docker image with gRPC 1.4

The new `1.4` folder contains a Dockerfile that
implements gRPC 1.4 for Python 3.6.

davidgasquez commented:
Thanks for the information @nathanielmanistaatgoogle! 😄 I updated the License on both _Dockerfiles_.

nathanielmanistaatgoogle commented:
Thanks!

davidgasquez commented:
Sorry @nathanielmanistaatgoogle, do you know if the new images will be available in https://hub.docker.com/r/grpc/python/builds/?

markmandel commented:
I can set that up - at an event at the moment, but will jump on it today/tomorrow. Feel free to harass me if need be. 👍

markmandel commented:
New images are now live!

davidgasquez commented:
Thanks for setting that up @markmandel! Also, thanks again @nathanielmanistaatgoogle for the help. 😄
